ux.symfony.com/src/Service/CookbookFactory.php

```diff
@@ -12,27 +12,65 @@
 namespace App\Service;
 
 use App\Model\Cookbook;
-use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use League\CommonMark\Environment\Environment;
+use League\CommonMark\Extension\CommonMark\CommonMarkCoreExtension;
+use League\CommonMark\Extension\FrontMatter\FrontMatterExtension;
+use League\CommonMark\Extension\FrontMatter\FrontMatterProviderInterface;
+use League\CommonMark\MarkdownConverter;
 
-class CookbookFactory
+final class CookbookFactory
 {
-    public function __construct(
-        private readonly CookbookParser $cookbookParser,
-        private readonly UrlGeneratorInterface $urlGenerator,
-    ) {
-    }
+    private MarkdownConverter $converter;
 
-    public function buildFromFile(\SplFileInfo $file): Cookbook
+    public function createFromFile(string $file): Cookbook
     {
-        $content = $file->getContents();
+        if (!file_exists($file)) {
+            throw new \InvalidArgumentException(\sprintf('File "%s" not found.', $file));
+        }
+
+        $content = file_get_contents($file);
+
+        $parsedContent = $this->getConverter()->convert($content);
+        if (!$parsedContent instanceof FrontMatterProviderInterface) {
+            throw new \RuntimeException('Unable to parse the cookbook.');
+        }
+
+        if (!\is_array($frontMatter = $parsedContent->getFrontMatter())) {
+            throw new \RuntimeException(\sprintf('Cookbook file "%s" does not contains Front Matter data.', $file));
+        }
+        if (!isset($frontMatter['title']) || !\is_string($frontMatter['title'])) {
+            throw new \RuntimeException('Missing title in Front Matter.');
```

Comment:
```suggestion
            throw new \RuntimeException('Front Matter data "title" is missing, or is not a string.');
```

Reply:
Same comments for the description/image/tags

Reply:
Should be a logical exception here to me, with no comments. I will refactor this later, so here is fine to me for now.